server/memphis_handlers_tags.go

```diff
@@ -0,0 +1,467 @@
+			return
+		}
+	}
+	c.IndentedJSON(200, []string{})
+}
+
+func (th TagsHandler) RemoveTags(c *gin.Context) {
+	var body models.RemoveTagsSchema
+	ok := utils.Validate(c, &body, false, nil)
+	if !ok {
+		return
+	}
+	for _, tagName := range body.Names {
+		name := strings.ToLower(tagName)
+		exist, tag, err := IsTagExist(name)
+
+		if err != nil {
+			serv.Errorf("RemoveTags error: " + err.Error())
+			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+			return
+		}
+		if !exist {
+			c.IndentedJSON(200, []string{})
+			return
+		}
+		switch body.EntityType {
+		case "station":
+			station_name, err := StationNameFromStr(body.EntityName)
+			if err != nil {
+				serv.Errorf("RemoveTags error: " + err.Error())
+				c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+				return
+			}
+			exist, station, err := IsStationExist(station_name)
+			if err != nil {
+				serv.Errorf("RemoveTags error: " + err.Error())
+				c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+				return
+			}
+			if !exist {
+				serv.Errorf("Station with this name does not exist")
+				c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+				return
+			}
+			_, err = tagsCollection.UpdateOne(context.TODO(), bson.M{"_id": tag.ID},
+				bson.M{"$pull": bson.M{"stations": station.ID}})
+			if err != nil {
+				serv.Errorf("RemoveTags error: " + err.Error())
+				c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+				return
+			}
+
+		// case "schema":
+		// 	exist, schema, err := IsSchemaExist(body.EntityName)
+		// 	if err != nil {
+		// 		serv.Errorf("RemoveTags error: " + err.Error())
+		// 		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		// 		return
+		// 	}
+		// 	if !exist {
+		// 		serv.Errorf("Schema with this name does not exist")
+		// 		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		// 		return
+		// 	}
+		// 	_, err = tagsCollection.UpdateOne(context.TODO(), bson.M{"_id": tag.ID},
+		// 		bson.M{"$pull": bson.M{"schemas": schema.ID}})
+		// 	if err != nil {
+		// 		serv.Errorf("RemoveTags error: " + err.Error())
+		// 		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		// 		return
+		// 	}
+		// case "user":
+		// 	exist, user, err := IsUserExist(body.EntityName)
+		// 	if err != nil {
+		// 		serv.Errorf("RemoveTags error: " + err.Error())
+		// 		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		// 		return
+		// 	}
+		// 	if !exist {
+		// 		serv.Errorf("User with this name does not exist")
+		// 		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		// 		return
+		// 	}
+		// 	_, err = tagsCollection.UpdateOne(context.TODO(), bson.M{"_id": tag.ID},
+		// 		bson.M{"$pull": bson.M{"users": user.ID}})
+		// 	if err != nil {
+		// 		serv.Errorf("RemoveTags error: " + err.Error())
+		// 		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		// 		return
+		// 	}
+		default:
+			serv.Errorf("RemoveTags error")
```

Comment:
again this is not an error it is an uinput mistake log with warn and return the user proper error message